[ch54607] Release/0.1

E-tanok opened the pull request.

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited files

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'plugin.json'

Commit:
Edited file 'plugin.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited files

Commit:
Edited files

Commit:
Edited files

Commit:
Edited files

Commit:
Edited files

Commit:
Edited files

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.json'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Created file '/python-lib'

Commit:
Created file 'python-lib/__init__.py'

Commit:
Created file 'python-lib/trigonometric_transformer.py'

Commit:
Edited file 'python-lib/trigonometric_transformer.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'python-lib/trigonometric_transformer.py'

Commit:
Created file 'python-lib/plugin_packages'

Commit:
Moved 'python-lib/__init__.py' to 'python-lib/plugin_packages'

Commit:
Moved 'python-lib/trigonometric_transformer.py' to 'python-lib/plugin_packages'

Commit:
Edited file 'python-lib/plugin_packages/__init__.py'

Commit:
Edited file 'python-lib/plugin_packages/trigonometric_transformer.py'

Commit:
Edited file 'python-lib/plugin_packages/__init__.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Moved 'python-lib/plugin_packages' to 'jython-processors'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Moved 'jython-processors/plugin_packages/__init__.py' to 'jython-processors/trigonometric-transformation'

Commit:
Moved 'jython-processors/plugin_packages/trigonometric_transformer.py' to 'jython-processors/trigonometric-transformation'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/trigonometric_transformer.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Removed 'jython-processors/trigonometric-transformation/__init__.py'

Commit:
Removed 'jython-processors/trigonometric-transformation/trigonometric_transformer.py'

Commit:
Removed 'jython-processors/plugin_packages'

Commit:
Removed 'python-lib'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited files

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

Commit:
Edited file 'jython-processors/trigonometric-transformation/processor.py'

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -2,13 +2,11 @@
 {
     "meta" : {
         // label: name of the data prep step as displayed, should be short
```
remove // comments to make it valid JSON

E-tanok replied:
Edit done.

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -2,13 +2,11 @@
-Then you naturally get the argument of this column values on this reversed circle.\
-Cosinus, sinus and tangent of this argument are computed to that you can intuitively analyse their values. This features take your phenomenon periodicity into account.\
-The way to read the measure on the resulting trigonometric circle is then the same as the one to read a clock (it starts from the top and goes in the clockwise direction of roatation )\
+        "description": "This processor applies a trigonometric transformation over a periodical column.\
```
summarize it to 140 characters (single line)

E-tanok replied:
Edite done : "This processor outputs features resulting from a trigonometric transformation over a periodical column." .

E-tanok replied:
Edit 2 : "This processor manages periodical columns (day of year, day of week ...) to create features by applying them a trigonometric transform".

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -45,15 +43,50 @@ The way to read the measure on the resulting trigonometric circle is then the sa
             "name": "periodical_column",
             "label": "periodiclal column",
```
```suggestion
            "label": "periodic column",
```

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -45,15 +43,50 @@ The way to read the measure on the resulting trigonometric circle is then the sa
             "name": "periodical_column",
```
```suggestion
            "name": "periodic_column",
```

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -45,15 +43,50 @@ The way to read the measure on the resulting trigonometric circle is then the sa
             "name": "periodical_column",
             "label": "periodiclal column",
             "type": "COLUMN",
```
The plugin should handle multiple columns 
```suggestion
            "type": "COLUMNS",
```

alexcombessie replied:
solved by #89233d4

E-tanok replied:
Edit done : We can now handle multiple columns.

E-tanok replied:
Edit : reversed back to "COLUMN" , after another discussion with you

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -45,15 +43,50 @@ The way to read the measure on the resulting trigonometric circle is then the sa
-            "description": "The period of the column containing the periodical information (ex: 365.24 for day of year, 30.43 for day of month, 7 for day of week,, 12 for month....)"
+            "description": "Periodical column period"
+        },
+        {
```
reduce UX complexity: cos, sin, tan should all be output. The user can use the remove column processor if need be,

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -45,15 +43,50 @@ The way to read the measure on the resulting trigonometric circle is then the sa
             "mandatory": true
         },
+        {
+            "name": "process_as_time_dimension",
```
Remove this parameter. We should handle this internally in the plugin code for valid date columns

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -2,13 +2,11 @@
 {
     "meta" : {
         // label: name of the data prep step as displayed, should be short
-        "label": "Custom processor polarize periodical feature",
```
general comment: periodic seems to be more "English" than periodical. I would replace everywhere in plugin identifiers and labels.

"Periodic feature engineering" looks like a good choice. It has 77 million search results on Google, which is more than other searches I have tried.

alexcombessie commented on jython-processors/trigonometric-transformation/processor.json:
```diff
@@ -45,15 +43,50 @@ The way to read the measure on the resulting trigonometric circle is then the sa
             "type": "COLUMN",
-            "description": "The column containing a periodical information (ex: day of year, day of month, day of week, month....)",
+            "description": "Column containing a periodical information",
             "mandatory": true
```
```suggestion
            "mandatory": true,
                        "allowedColumnTypes": [
                "date",
                "int",
                "double",
                "float"
            ]
```

alexcombessie commented on jython-processors/trigonometric-transformation/processor.py:
```diff
@@ -1,20 +1,117 @@
+import math
+
+#Class definition :
+class TrigonometricTransformer():
```
move classes to module in python-lib

alexcombessie commented on jython-processors/trigonometric-transformation/processor.py:
```diff
@@ -1,20 +1,117 @@
+        pass
+    
+    def build_trigonometric_transformer(self, L_choosed_functions):
+        cos_choosed = L_choosed_functions[0]
```
compute all of them by default

alexcombessie commented on jython-processors/trigonometric-transformation/processor.py:
```diff
@@ -1,20 +1,117 @@
+            cos_val, sin_val, tan_val = trigonometric_transformer.compute_trigonometric_transform(arg_val)
+        log_message = None
+        
+    except Exception as e:
```
Always catch specifics exception such as ValueError

alexcombessie commented on jython-processors/trigonometric-transformation/processor.py:
```diff
@@ -1,20 +1,117 @@
+        to the functions set by the build_trigonometric_transformer() method.
+    - 'compute_argument' : is a methiod allowing to convert a numerical value from a 
+        periodical feature  into a trigonometric argument. 
+    - 'compute_trigonometric_features' : is a method allowing, if 'process_as_time_dimension==True', to 
```
What is the % time saving benefit of pre-computation? Can you benchmark it compared to a simple approach?

alexcombessie commented on jython-processors/trigonometric-transformation/processor.py:
```diff
@@ -1,20 +1,117 @@
-    cos_val = math.cos(arg_val)
-    sin_val = math.sin(arg_val)
-    tan_val = math.tan(arg_val)
+    L_values = [arg_val, cos_val, sin_val, tan_val]
```
use valid english variables names following this_casing_example.
```suggestion
    output_values = [arg_val, cos_val, sin_val, tan_val]
```

E-tanok replied:
Edit done : variables updated everywhere in the code.

E-tanok replied:
Edit : The code structure changed : we do not use anymore this variable

alexcombessie commented on plugin.json:
```diff
@@ -10,7 +10,10 @@
         "label": "Trigonometric transformation",
 
         // description: longer string to help end users understand what this plugin does
```
remove // comments

E-tanok replied:
Edit done.

alexcombessie commented on plugin.json:
```diff
@@ -10,7 +10,10 @@
         "label": "Trigonometric transformation",
```
```suggestion
        "label": "Periodic feature engineering",
```

E-tanok replied:
Edit done.

alexcombessie commented on plugin.json:
```diff
@@ -10,7 +10,10 @@
 
         // description: longer string to help end users understand what this plugin does
-        "description": "",
+        "description": "This plugin adds a processor in the prepare recipe : 'Custom processor trigonometric transformation'.\
```
summarize in a single line / 140 characters 
```suggestion
        "description": "Extract features from periodic columns (day of the week, latitude/longitude, etc.)"
```

E-tanok replied:
Edit done.

alexcombessie commented on jython-processors/trigonometric-transformation/processor.py:
```diff
@@ -1,20 +1,117 @@
+        of its cos/sin/tan.
+    """
+    def __init__(self, L_choosed_functions, period, process_as_time_dimension, range_period):
+        self.L_choosed_functions = L_choosed_functions
```
use valid english variables names following this_casing_example.

E-tanok replied:
Edit done.

alexcombessie reviewed (changes requested):
Thanks for the PR @E-tanok! I have left comments on code readability / simplification.
I have a big question related to precomputation, which adds some complexity. Is it justified by a measurable time saving?

alexcombessie committed:
Update jython-processors/trigonometric-transformation/processor.json

alexcombessie commented:
Hello @E-tanok !
Hope all is well. Do you know if you'll have some time to finalize this over the summer?
Cheers,
Alex

ghost commented:
Please update plugin id too, periodic-feature-engineering works.

Commit:
MODIFICATIONS :
1 : All remarks (https://github.com/dataiku/dss-plugin-periodic-feature-engineering/pull/3) taken into account.
2 : Docstring and comments added on the class + the main code

alexcombessie commented:
Clubhouse integration: [ch54607](https://app.clubhouse.io/dataiku/story/54607/initial-release)

alexcombessie committed:
start refactoring

alexcombessie committed:
simplify UX

alexcombessie committed:
Update plugin.json

Commit:
MODIFICATIONS :

- processor.json : 
--> switched from "COLUMNS" to "COLUMN".
--> UX simplified (less parameters). Now we just set
-----> The column where we apply the computation
-----> The feature period's min and max values

- processor.py : 
--> Code simplified : only the "process" function

Commit:
processor.json  : 
- Naming simplifications : parameters "Period min/max" replaced by "Minimum/Maximum value"
- Descriptions removed on Minimum/Maximum value parameters

processor.py : 
- More exceptions catched

Commit:
descriptions deletion in parameters "Minimum/Maximum value" to be aligned with parameter "Periodic column".

Commit:
The values are now min/max scaled

Commit:
The log message is now contextualized by the input column name

alexcombessie commented:
@E-tanok what is the status of this PR? Let me know if this is ready for review :)

alexcombessie commented:
Ping @E-tanok Could you help finalize this?
